lib/ansible/modules/cloud/vmware/vmware_host_kernel_manager.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2019, Aaron Longchamps, <a.j.longchamps@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_host_kernel_manager
+short_description: Configure kernel module options on an ESXi host or cluster.
+description:
+>-
+    This module can be used to configure kernel module options on a single ESXi host or a cluster.
+    Users can specify an ESXi hostname or cluster name. All connected ESXi hosts in scope will be
+    configured when specified. If a host is not connected at time of configuration, it will be marked
+    as such in the output and be skipped. Kernel module options may require a reboot to take effect
+    which is not covered by this script.
+version_added: '2.8'
+author:
+- Aaron Longchamps (@alongchamps)
+notes:
+- Tested on vSphere 6.0
+requirements:
+- python >= 2.7
+- PyVmomi
+options:
+  esxi_hostname:
+    description:
+    - Name of the ESXi host to work on
+    - This parameter is required if C(cluster_name) is not specified
+    type: str
+  cluster_name:
+    description:
+    - Name of the VMware cluster to work on
+    - All ESXi hosts in this cluster will be configured
+    - This parameter is required if C(esxi_hostname) is not specified
+    type: str
+  kernel_module_name:
+    description:
+    - Name of the kernel module to be configured
+    required: true
+    type: str
+  kernel_module_option:
+    description:
+    - Specified configurations will be applied to the given module
+    - These values are specified in key=value pairs and separated by a space when there are multiple options
+    required: true
+    type: str
```

Comment:
how about creating  a list of key and value pairs 

```yaml
- name: Configure kernel module parameters
  vmware_host_kernel_manager:
    hostname: '{{ vcenter_hostname }}'
    username: '{{ vcenter_username }}'
    password: '{{ vcenter_password }}'
    esxi_hostname: '{{ esxi_hostname }}'
    kernel_modules:
      - "tcpip4": "ipv6=0"
      - "vmw_psp_rr": "maxPathsPerDevice=2"
```

Reply:
Talked with akasurde on IRC, for now I'm going to leave it as kernel_module_name and kernel_module_option because it can be looped over with loop controls built in to ansible already.